compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -65,93 +41,121 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleReshape *node)
 namespace sinf
 {
 
+/**
+ * @note  CircleReshape always has two inputs: `tensor` and `shape`.
+ *        The `shape` input can be CircleConst, CircleOutputDummy, or CircleNode.
+ *        - If the `shape` input is CircleConst, the shape is inferred from the constant.
+ *        - If the `shape` input is CircleOutputDummy, the shape is inferred from
+ *          the attribute if it exists. If the attribute does not exist,
+ *          the shape is inferred from the node iteself.
+ *        - If the `shape` input is CircleNode, the dynamic shape is propagated.
+ */
```

Comment:
Q: Is there an easy way to apply these changes one by one? 

By reading diff page (https://github.com/Samsung/ONE/pull/13999/files?diff=unified#diff-53743de477b450a036961928b169bb3cdbdfa63c96717ac0f9d88a5b3f696338R45-R52), It is hard to distinguish how each logics are updated.

Reply:
IDK this is possible or not, but I'm trying to split changes into smalls : 

e.g. 
	* support CircleConst, not CircleConst case first
	* apply should_infer 
	* support CircleDummy next